tsdb/shard.go

```diff
@@ -66,18 +66,39 @@ type Shard struct {
 	LogOutput io.Writer
 }
 
+// ShardLocation is used when calling NewShard to specify the shard's
+// database, retention policy, and location of files on disk.
+type ShardLocation struct {
+	// Name of the database this shard belongs to
+	Database string
+
+	// Name of the retention policy this shard belongs to
+	RetentionPolicy string
+
+	// Path to this shard's location on disk
+	Path string
+
+	// Path to this shard's WAL location
+	WALPath string
+}
+
```

Comment:
Initially, `NewShard` accepted `path` and `walPath` as distinct arguments, but that information was insufficient to tag the Shard's stats with the `Database` and `RetentionPolicy`. I thought the API for `NewShard` was cleaner to accept a `ShardLocation`, for a total of 4 arguments, rather than passing in the "location" via 4 separate strings, which would have brought it up to 7 arguments.

I don't feel especially strong about it, and there's a merge conflict to fix anyway, so I can remove the `ShardLocation` type if that's what you prefer @benbjohnson.